src/Symfony/Component/Yaml/Inline.php

```diff
@@ -406,7 +416,10 @@ private static function parseMapping($mapping, &$i = 0, $references = array())
                         // Spec: Keys MUST be unique; first one wins.
                         // Parser cannot abort this mapping earlier, since lines
                         // are processed sequentially.
-                        if (!isset($output[$key])) {
+                        // But overwriting is allowed when a merge node is used in current block.
+                        if ('<<' === $key) {
+                            $output += $value;
+                        } elseif ($allowOverwrite || !isset($output[$key])) {
```

Comment:
These conditions are a bit hard to understand because above you have this: `if ('<<' === $key) { $allowOverwrite = true; }`. So, `if('<<' === $key)` and `if($allowOverwrite)` are interchangeable and this `if ... elseif` is a bit confusing.

Reply:
This part is executed several times (for each entry in the mapping). So `$allowOverwrite` can be `true` in a subsequent iteration where `$key` is different from `<<`.

Reply:
@javiereguiluz Is there still anything you would like to see changed?